lib/types.nix

```
@@ -605,6 +609,13 @@ let
         };
       };
 
+      fileset = mkOptionType {
+        name = "fileset";
+        description = "fileset";
+        check = isFileset;
+        merge = loc: defs: unions (map (x: x.value) defs);
+      };
```

```suggestion
        emptyValue.value = fileset.empty;
      };
```

Test case:
```
"${fileset.toSource { fileset = (evalModules { modules = [ { options.foo = mkOption { type = types.lazyAttrsOf types.fileset; }; config.foo.bar = mkIf false throw; } ]; }).config.foo.bar; root = ./.; }}" == "${builtins.path { filter = _: _: false; name = "source"; path = ./.; }}"
```

It'd be good to have some test cases in `lib/tests/modules.sh` and/or `lib/tests/misc.nix`.

---

I have added module tests in d4d4c54bb018014ba4b8d0864537a31202eeb638. The bug you fixed here would be caught by `ok5`. I'll let you close the conversation if you are satisfied. I am happy to add more stuff if you think that is useful or necessary!